source/common/quic/platform/flags_impl.cc

```diff
@@ -53,7 +54,12 @@ void FlagRegistry::resetFlags() const {
   }
 }
 
-Flag* FlagRegistry::findFlag(const std::string& name) const {
+Flag* FlagRegistry::findFlag(absl::string_view name) const {
+  if (name.substr(0, ConstantQuicFlagPrefix.length()) == ConstantQuicFlagPrefix) {
+    // Convert the envoy reloadable flag to quiche flag.
+    name = absl::StrCat("FLAGS_quic_reloadable_flag_",
+                        name.substr(ConstantQuicFlagPrefix.length() - 4));
```

Comment:
This is doing string copy. Can we twinkle  the flag name so that we can use the passed in name or substring to search in the map?

Reply:
The Envoy runtime will pass in names like "envoy.reloadable_features.quic_foo". If we want to do substring without copying, then in out flag registry, the flag will look like "quic_foo". 
This way we hide out the reloadable/restart/protocol attribute from the flag name.
I mean our flag system will still work without all the prefixes, but I wanna make sure people are ok with that.

Reply:
Can we make quic runtime feature name to be "envoy.reloadable_features.FLAGS_quic_reloadable_flag_quic_foo"?  As runtime overloading can only work for reloadable flags, it's good to include "quic_reloadable" in the runtime feature name.
TypedFlag::name_ is also defined by ourselves. So we can modify it as well.

Reply:
Good idea. Now the envoy override of quic flags will be of form "envoy.reloadable_features.FLAGS_quic_reloadable_flag_quic_foo".
This way we avoid doing strcat() in findFlag().